cpp/src/join/mixed_join_kernels_semi_compound.cu

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2022-2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "join/mixed_join_common_utils.cuh"
+#include "join/mixed_join_kernel_semi_impl.cuh"
+
+namespace cudf {
+namespace detail {
+template __global__ void mixed_join_semi<DEFAULT_JOIN_BLOCK_SIZE, true, row_hash_no_nested>(
```

Comment:
We aren't supposed to have raw `__global__` attributes anymore, in my understanding. Let's try this through this PR.
```suggestion
template CUDF_KERNEL void mixed_join_semi<DEFAULT_JOIN_BLOCK_SIZE, true, row_hash_no_nested>(
```